processing/src/main/java/io/druid/segment/StorageAdapter.java

```diff
@@ -46,4 +49,10 @@
   public ColumnCapabilities getColumnCapabilities(String column);
   public int getNumRows();
   public DateTime getMaxIngestedEventTime();
+
+  public abstract static class AbstractStorageAdapter implements StorageAdapter {
```

Comment:
I'm not sure I understand why this was created.  It looks like it is overriding the 3-argument implementation to delegate to the 4-argument implementation of `makeCursors`.

I believe it's being done as an attempt at maintaining backwards compatibility while adding the `descending` argument.  I think that in general, we should approach adding the parameter in a different manner.  Perhaps as part of the `QueryGranularity` or something else.  I'm going to write a top-level comment about this more.